### PR TITLE
Gentoo fixes

### PR DESCRIPTION
**Pull request by prometheanfire**
Was more explicit on eselect usage, which can vary over time.  I assumed you wanted what I selected as it's what users generally want.  Would be nice not to need python2.7 though :P

I also added better emerge / make options for build speed and completeness

Last thing I did was update the cloud init version.

If you need me to ammend or remove any of these commits, let me know.

### CODE REVIEW

**Comment by naterh** [comment]
Python 2.7 is a hard req for cloud-init and that isn't really changing anytime soon.  The upstream project has just started to add changes for 3 compatibility but, it isn't fully tested.  Nor is python 3 with the package dependencies which likely would fail too.

Which better openstack support are you referring to in cloud-init 0.7.6?  Looking at the source, not much has changed in the gentoo distro class since I initially added it.  The ConfigDrive data source sources module hasn't changed much either other than removing the default date version which is how we still use ConfigDrive as part of our current Nova config in Public Cloud.  The changelog comments around better OpenStack support for cloud-init on the launchpad project was referring to adding the OpenStack datasource which uses the newly added metadata service and is something that we don't currently have in our cloud atm.

File: Gentoo_PVHVM-2.sh [hunk 1]
```diff
@@ -79,6 +79,7 @@ EOF
 # misc portage installs
 emerge iproute2
 emerge vim
+eselect editor set /usr/bin/vi
```
Comment (naterh) [inline Gentoo_PVHVM-2.sh]: We don't set default editors in any of our images.

**Comment by prometheanfire** [comment]
I was just looking at the changelog for 0.7.6.  It's more of an update.

Let me know if all you need from me is removing the eselect for vi or if you need the cloud-init change removed as well.

File: Gentoo_PVHVM-2.sh [hunk 2]
```diff
@@ -40,7 +40,7 @@ EOF
 dd if=/dev/zero of=/swapfile1 bs=1024 count=1048576 && mkswap /swapfile1 && chown root:root /swapfile1 && chmod 0600 /swapfile1 && swapon /swapfile1
 emerge-webrsync
 emerge --sync --quiet
-emerge -u world
+emerge -uDN --quiet-build --jobs=5 --with-bdeps=y @world
```
Comment (naterh) [inline Gentoo_PVHVM-2.sh]: I would rather not do quiet-build with pkg installs for console logging etc.

Comment (prometheanfire) [inline Gentoo_PVHVM-2.sh]: That gets rid of jobs too.  Would it be better to set a log and ship the logs when done?

**Comment by naterh** [comment]
If you drop the default editor, --quiet-build and cloud-init 0.7.6 we should be good.  Thanks for your changes!

File: Gentoo_PVHVM-2.sh (hunk 2 above)
Comment (naterh) [inline Gentoo_PVHVM-2.sh]: to standardize across all the image builds we're grabbing the hypervisor console logs.

**Comment by prometheanfire** [comment]
ok, updated

**Comment by naterh** [comment]
Thanks!

File: Gentoo_PVHVM-2.sh [hunk 3]
```diff
@@ -40,7 +40,7 @@ EOF
 dd if=/dev/zero of=/swapfile1 bs=1024 count=1048576 && mkswap /swapfile1 && chown root:root /swapfile1 && chmod 0600 /swapfile1 && swapon /swapfile1
 emerge-webrsync
 emerge --sync --quiet
-emerge -u world
+emerge -uDN --with-bdeps=y @world
```
Comment (naterh) [inline Gentoo_PVHVM-2.sh]: @prometheanfire This line change has added 220MB to the base image and also increased the build time from 45m to 135m or 300% and has been reverted.  It pulled in a lot of perl pkgs that are not needed.

**Pull request merged**